src/pykabomu/rina/qpc01.py

```diff
@@ -0,0 +1,41 @@
+from typing import Optional
+from pykabomu.common.abstractions.bytequeue import *
+from abc import ABC, abstractmethod
+
+from pykabomu.common.abstractions.intqueue import *
+from pykabomu.common.exceptions import KabomuException
+from pykabomu.common.helpers import byteutils
+
+# traffic regulator abstract class
+class TrafficRegulator(ABC):
+
+    def __init__(self, begin_send, begin_receive):
+        self.begin_send = begin_send
+        self.begin_receive = begin_receive
+
+# declare public method begin_send
+    def begin_send(self, byteQueue, sendCallback):
+        pass
+
+# declare public method begin_receive
+    def begin_receive(self, byteQueue):
+        pass
```

Comment:
This is where the second part of the implementation should have gone.
```suggestion
        self.sendCallback(byteQueue)
        self.sendCallback = None
```